seloger-scraper/seloger.py

```diff
@@ -0,0 +1,99 @@
+"""
+This is an example web scraper for seloger.com.
+
+To run this scraper set env variable $SCRAPFLY_KEY with your scrapfly API key:
+$ export $SCRAPFLY_KEY="your key from https://scrapfly.io/dashboard"
+"""
+import os
+import json
+from scrapfly import ScrapeConfig, ScrapflyClient, ScrapeApiResponse
+from typing import Dict, List
+from pathlib import Path
+from loguru import logger as log
+
+SCRAPFLY = ScrapflyClient(key=os.environ["SCRAPFLY_KEY"])
+
+BASE_CONFIG = {
+    "asp": True,
+    "country": "fr",
+}
+
+output = Path(__file__).parent / "results"
+output.mkdir(exist_ok=True)
+
+
+def parse_search(result: ScrapeApiResponse):
+    """parse property listing data from seloger search pages"""
+    # select the script tag from the HTML
+    script_jsons = result.selector.xpath("//script[contains(., 'window[\"initialData\"]')]/text()").re(
+        r'JSON.parse\("(.+?)"\)'
+    )
+    # decode the JSON data
+    datasets = [json.loads(script_json.encode("utf-8").decode("unicode_escape")) for script_json in script_jsons]
+    cards = []
+    # validate the cards data to avoid advertisement cards
+    for card in datasets[0]["cards"]["list"]:
+        if len(card) > 12:
+            cards.append(card)
+    search_meta = datasets[0]["navigation"]
+    return {"results": cards, "search": search_meta}
+
+
+def _max_search_pages(search_meta: Dict) -> int:
+    """get the maximum number of pages available on the search"""
+    return search_meta["counts"]["count"] // search_meta["pagination"]["resultsPerPage"]
+
+
+def parse_property_page(result: ScrapeApiResponse):
+    """parse property data from the nextjs cache"""
+    # select the script tag from the HTML
+    next_data = result.selector.css("script[id='__NEXT_DATA__']::text").get()
+    listing_data = json.loads(next_data)["props"]["pageProps"]["listingData"]
+    # extract property data from the property page
+    property_data = {}
+    property_data["listing"] = listing_data["listing"]
+    property_data["agency"] = listing_data["agency"]
+    return property_data
+
+
+async def scrape_search(
+    url: str,
+    scrape_all_pages: bool,
+    max_pages: int = 10,
+) -> List[Dict]:
+    """scrape seloger search page"""
```

Comment:
can we add exmaple search URL so people can understand the function faster? Same for `scrape_property`